apps/status-api/src/controllers/OracleStatusController.ts

```diff
@@ -0,0 +1,42 @@
+import { Controller, Get, Query } from '@nestjs/common'
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { Oracle } from '@defichain/whale-api-client/dist/api/Oracles'
+import { SemaphoreCache } from '../../../whale/src/module.api/cache/semaphore.cache'
+
+@Controller('/status')
+export class OracleStatusController {
+  constructor (
+    private readonly client: WhaleApiClient,
+    protected readonly cache: SemaphoreCache
+  ) {
+  }
+
+  @Get('oracles')
+  async getOracleStatus (
+    @Query('address') oracleAddress: string
+  ): Promise<{ [key: string]: string }> {
+    const oracle: Oracle = await this.cachedGet('ORACLE', async () => {
```

Comment:
```suggestion
    const oracle: Oracle = await this.cachedGet(`oracle-${oracleAddress}`, async () => {
```
Cache id has to be unique to the oracle.